### PR TITLE
Removing some Dead Code in Discovery

### PR DESCRIPTION
**Pull request by original-brownbear**
* Just a few obviously unused things I found while researching other things

### CODE REVIEW

**Comment by andrershov** [comment]
@original-brownbear LGTM, if it has any sense after fixing merge conflicts.

**Review by andrershov — APPROVED** [review]

**Comment by original-brownbear** [comment]
@andrershov thanks! But unfortunately, it looks like the diff has exactly `0` lines after merging :D

**Pull request closed without merging**
